src/api/time-scale-api.ts

```
@@ -66,6 +60,18 @@ export class TimeScaleApi implements ITimeScaleApi, IDestroyable {
 		this._model.setTargetTimeRange(convertedRange);
 	}
 
+	public getVisibleIndexRange(): TimePointIndexRange | null {
+		const indexRange = this._timeScale().visibleIndexRange();
+		return indexRange ? clone(indexRange) : null;
+	}
+
+	public setVisibleIndexRange(range: TimePointIndexRange): void {
+		if (range.from > range.to) {
```

what do you think about asserting here instead?

---

Good point! I've changed it to `assert()`. It also makes more sense to me, since if you end up calling this with an invalid range, it's probably more useful to be notified of that, rather than silently ignoring it.